internal/graph/cursors.go

```diff
@@ -65,84 +55,57 @@ func (ci cursorInformation) withClonedLimits() cursorInformation {
 		currentCursor:          ci.currentCursor,
 		outgoingCursorSections: ci.outgoingCursorSections,
 		limits:                 ci.limits.clone(),
-		revision:               ci.revision,
 	}
 }
 
-// hasHeadSection returns true if the current cursor has the given name as the prefix of the cursor.
-func (ci cursorInformation) hasHeadSection(name string) (bool, error) {
-	if ci.currentCursor == nil || len(ci.currentCursor.Sections) == 0 {
-		return false, nil
-	}
-
-	if ci.currentCursor.Sections[0] != name {
-		return false, spiceerrors.MustBugf("expected cursor section %s in %v", name, ci.currentCursor.Sections)
+// headSectionValue returns the string value found at the head of the incoming cursor.
+// If the incoming cursor is empty, returns empty.
+func (ci cursorInformation) headSectionValue() (string, bool) {
```

Comment:
I feel like we could easily have a generic function for head of a slice and the caller easily just use that instead of having this specialized function